pkg/service/backup/worker_upload.go

```diff
@@ -71,6 +71,14 @@ func (w *worker) uploadHost(ctx context.Context, h hostInfo) error {
 			err = errors.Wrap(w.deleteTableSnapshot(ctx, h, d), "delete table snapshot")
 		}()
 
+		if ok, err := w.useScyllaBackupAPI(ctx, d, h); err != nil {
+			return errors.Wrap(err, "check Scylla backup API support")
```

Comment:
Why return on error, and not failback to RClone API ?

Reply:
We could do it, but do we want to?
Because of https://github.com/scylladb/scylla-manager/issues/4163 it will be possible to control which API should be primarily used for the backup process.
The fallback without any error analysis can prolong the agony of a problematic backup (which might fail with Rclone API as well).
There is also a question of whether interrupting backup with Scylla and resuming it with Rclone is safe (although I don't know why wouldn't it be).

Reply:
Thanks for the context.
Let's not failback.